framework/include/utils/ParallelUniqueId.h

```diff
@@ -32,20 +37,30 @@ class ParallelUniqueId
   ParallelUniqueId()
   {
 #ifdef LIBMESH_HAVE_TBB_API
-    ids.pop(id);
-#else
-#ifdef LIBMESH_HAVE_PTHREAD
-    id = Threads::pthread_unique_id();
+    _ids.pop(id);
+#elif LIBMESH_HAVE_OPENMP
+    id = omp_get_thread_num();
+#elif LIBMESH_HAVE_PTHREAD
+    Threads::spin_mutex::scoped_lock lock(_pthread_id_mutex);
+
+    if (_ids.empty())
+      throw MooseException("No Thread IDs available in ParallelUniqueID. Did you forget to initialize()?");
+
+    id = _ids.front();
+    _ids.pop();
```

Comment:
@dschwen Yes, that's the design.  Each thread is supposed to get and hold ONE tid.  This has been the design for 7 years without issue ;-)

This object is NOT, at all, anything that should EVER be used by a user (or even an advanced developer :-).  It is strictly to be used in threaded looping classes in the framework.  Every other object is fed a "tid" and has no need to interact with this.